README.yaml

```diff
@@ -0,0 +1,163 @@
+---
+#
+# This is the canonical configuration for the `README.md`
+# Run `make readme` to rebuild the `README.md`
+#
+
+# Name of this project
+name: terraform-aws-dynamic-subnets
+
+# Logo for this project
+#logo: docs/logo.png
+
+# License of this project
+license: "APACHE2"
+
+# Canonical GitHub repo
+github_repo: cloudposse/terraform-aws-dynamic-subnets
+
+# Badges to display
+badges:
+  - name: "Build Status"
+    image: "https://travis-ci.org/cloudposse/terraform-aws-dynamic-subnets.svg?branch=master"
+    url: "https://travis-ci.org/cloudposse/terraform-aws-dynamic-subnets"
+  - name: "Latest Release"
+    image: "https://img.shields.io/github/release/cloudposse/terraform-aws-dynamic-subnets.svg"
+    url: "https://github.com/cloudposse/terraform-aws-dynamic-subnets/releases/latest"
+  - name: "Slack Community"
+    image: "https://slack.cloudposse.com/badge.svg"
+    url: "https://slack.cloudposse.com"
+
+related:
+  - name: "terraform-aws-vpc"
+    description: "Terraform Module that defines a VPC with public/private subnets across multiple AZs with Internet Gateways"
+    url: "https://github.com/cloudposse/terraform-aws-vpc"
+  - name: "terraform-aws-vpc-peering"
+    description: "Terraform module to create a peering connection between two VPCs"
+    url: "https://github.com/cloudposse/terraform-aws-vpc-peering"
+  - name: "terraform-aws-kops-vpc-peering"
+    description: "Terraform module to create a peering connection between a backing services VPC and a VPC created by Kops"
+    url: "https://github.com/cloudposse/terraform-aws-kops-vpc-peering"
+  - name: "terraform-aws-named-subnets"
+    description: "Terraform module for named subnets provisioning."
+    url: "https://github.com/cloudposse/terraform-aws-named-subnets"
+  - name: "terraform-aws-kops-metadata"
+    description: "Terraform module for named subnets provisioning."
+    url: "https://github.com/cloudposse/terraform-aws-kops-metadata"
+
+# Short description of this project
+description: |-
+  Terraform module to provision public and private [`subnets`](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_Subnets.html) in an existing [`VPC`](https://aws.amazon.com/vpc)
+
+  __Note:__ this module is intended for use with an existing VPC and existing Internet Gateway.
+  To create a new VPC, use [terraform-aws-vpc](https://github.com/cloudposse/terraform-aws-vpc) module.
+
+# How to use this project
+usage: |-
+  ```hcl
+  module "subnets" {
+    source              = "git::https://github.com/cloudposse/terraform-aws-dynamic-subnets.git?ref=master"
+    namespace           = "cp"
+    stage               = "prod"
+    name                = "app"
+    region              = "us-east-1"
+    vpc_id              = "vpc-XXXXXXXX"
+    igw_id              = "igw-XXXXXXXX"
+    cidr_block          = "10.0.0.0/16"
+    availability_zones  = ["us-east-1a", "us-east-1b"]
+  }
+  ```
+  ## Subnet calculation logic
```

Comment:
Move this section to `docs/design.md` and the add `docs/design.md` to the `includes` section.